db/queries/assessment_records/queries.py

```diff
@@ -671,3 +674,76 @@ def get_assessment_records_by_round_id(round_id):
             )
 
     return output
+
+
+def associate_assessment_tags(application_id, tags: List):
+    existing_associated_tags = TagAssociation.query.filter(
+        TagAssociation.application_id == application_id,
+        TagAssociation.associated == True,  # noqa: E712
+    ).all()
+
+    # Create a dictionary to quickly lookup existing tags by tag_id
+    existing_associated_tags_dict = {
+        str(tag.tag_id): tag for tag in existing_associated_tags
+    }
+
+    # Iterate over the provided tags and update the tag associations
+    for tag in tags:
+        tag_id = tag.get("id")
+        user_id = tag.get("user_id")
+        associated = True  # Set associated value to True for provided tags
+
+        # Check if the tag already exists in the database
+        if tag_id in existing_associated_tags_dict.keys():
+            pass
+        else:
+            # Create a new tag association
+            new_tag = TagAssociation(
+                application_id=application_id,
+                tag_id=tag.get("id"),
+                associated=associated,
+                user_id=user_id,
+            )
+            db.session.add(new_tag)
```

Comment:
Could we just use `not in` rather than an `else:`?

Also `in` via dict implicitly checks keys, so `.keys()` is unnecessary

```python
>>> 'a' in {'a':1}
True
```

```suggestion
        if tag_id not in existing_associated_tags_dict:
            # Create a new tag association
            new_tag = TagAssociation(
                application_id=application_id,
                tag_id=tag.get("id"),
                associated=associated,
                user_id=user_id,
            )
            db.session.add(new_tag)
```